tiledb/sm/misc/utils.cc

```diff
@@ -469,14 +472,16 @@ template <>
 Status check_template_type_to_datatype<uint8_t>(Datatype datatype) {
```

Comment:
This function isn't called anywhere. Might as well remove it in this PR instead of altering it.